app.py

```diff
@@ -1,4 +1,4 @@
-from flask import Flask, redirect, render_template
+from flask import Flask, redirect, render_template, request, abort
```

Comment:
It seems that there is a slight conflict between the request import being before abort, as the main branch currently does not include it. After that is fixed I believe it will be ready to merge, as the rest of the code looks great to me.